src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -812,13 +812,40 @@ object Desugar {
 
           case b: PBlock => unit(blockD(ctx)(b))
 
-          case l: PLabeledStmt =>
+          case l: PLabeledStmt => {
             val proxy = labelProxy(l.label)
-            for {
-              _ <- declare(proxy)
-              _ <- write(in.Label(proxy)(src))
-              s <- goS(l.stmt)
-            } yield s
+            if (l.stmt.isInstanceOf[PForStmt]) {
```

Comment:
We can use match instead of using the pattern `isInstanceOf` followed by `asInstanceOf`